### DISCUSSION TITLE
Is there any way to determine whether a DataFetcher is default DataFetcher

### DISCUSSION THREAD

**Discussion by dugenkui03** [graphql-java/graphql-java#2754] [Q&A]
Is there any way to determine whether a DataFetcher is default DataFetcher

**Comment by bbakerman** [comment]
`instanceof PropertyDataFetcher` will be close - by default this the the default.

The default data fetch is this `defaultDataFetcherFactory = env -> PropertyDataFetcher.fetching(env.getFieldDefinition().getName());` in the code registry

But some one could change it so....

**Reply by bbakerman** [reply]
Can I ask what you want to achieve?

**Reply by dugenkui03** [reply]
@bbakerman Thanks for your response.

I find a bug in [spring graphql AutoRegistrationTypeVisitor](https://github.com/spring-projects/spring-graphql/issues/244) and try to fix it in [pr 302](https://github.com/spring-projects/spring-graphql/pull/302/files). The key point is how to determine whether a DataFetcher is dafault DataFetcher, but I don't know is this a correct way:

```java
private boolean hasDefaultDataFetcher(
			GraphQLCodeRegistry.Builder registry, GraphQLFieldsContainer parent,
			GraphQLFieldDefinition fieldDefinition) {

		DataFetcherFactoryEnvironment dataFetcherFactoryEnvironment = DataFetcherFactoryEnvironment
				.newDataFetchingFactoryEnvironment()
				.fieldDefinition(fieldDefinition).build();
		DataFetcher<?> defaultDataFetcher = registry.getDefaultDataFetcherFactory().get(dataFetcherFactoryEnvironment);
		DataFetcher<?> fetcher = registry.getDataFetcher(parent, fieldDefinition);
		return (fetcher != null &&
				(defaultDataFetcher != null && !(defaultDataFetcher.getClass().isInstance(fetcher)))
		);
	}
```
